python-lib/dku_io_utils/partitions_handling.py

```diff
@@ -16,15 +16,46 @@ def get_partition_root(dataset):
     dku_flow_variables = dataiku.get_flow_variables()
     file_path_pattern = dataset.get_config().get("partitioning").get("filePathPattern", None)
 
-    dimensions, types = get_dimensions(dataset)
+    dimensions, types = get_dataset_dimensions(dataset)
     partitions = get_partitions(dku_flow_variables, dimensions)
     file_path = complete_file_path_pattern(file_path_pattern, partitions, dimensions, types)
     file_path = complete_file_path_time_pattern(dku_flow_variables, file_path)
 
     return file_path
 
 
-def get_dimensions(dataset):
+def get_folder_partition_root(folder, is_source=False):
+    """Retrieve the partition root path using a dataiku.Folder.
+
+    Args:
+        dataset (dataiku.Dataset): Input or output dataset of the recipe used to retrieve the partition path pattern.
+        Boolean:  True if the folder must be considered as a source, False if destination
+
+    Returns:
+        Partition path or None if dataset is not partitioned.
+    """
+    folder_id = folder.get_id()
+    source = folder_id if is_source else None
+    dku_flow_variables = dataiku.get_flow_variables()
+    client = dataiku.api_client()
+    project = client.get_project(dataiku.default_project_key())
+    folder = project.get_managed_folder(folder_id)
+    folder.get_definition()["partitioning"]
+    folder_config = folder.get_definition()
+    file_path_pattern = folder_config.get("partitioning").get("filePathPattern", None)
+    dimensions, types = get_dimensions(folder_config)
+    partitions = get_partitions(dku_flow_variables, dimensions, source=source)
+    file_path = complete_file_path_pattern(file_path_pattern, partitions, dimensions, types)
+    file_path = complete_file_path_time_pattern(dku_flow_variables, file_path, source=source)
+    return file_path
+
+
+def get_dataset_dimensions(dataset):
```

Comment:
I am not a big fan of small functions which redirect to other bigger functions. There is a lot of overlap between `get_dataset_dimensions` and `get_dimensions`. I would rather have a single function called `get_dataset_dimensions`.